wagtail/admin/views/page_privacy.py

```diff
@@ -7,7 +7,7 @@
 
 
 def set_privacy(request, page_id):
-    page = get_object_or_404(Page, id=page_id)
+    page = get_object_or_404(Page, id=page_id).specific
```

Comment:
Nit: This can be optimised, since we only need the class instance, not all the fields on it
```suggestion
    page = get_object_or_404(Page, id=page_id).specific_deferred
```